pandas/_libs/index.pyx

```diff
@@ -98,6 +98,20 @@ cdef ndarray _get_bool_indexer(ndarray values, object val, ndarray mask = None):
     return indexer.view(bool)
 
 
+cdef _maybe_resize_array(ndarray values, Py_ssize_t loc, Py_ssize_t max_length):
+    """
+    Resize array if loc is out of bounds.
+    """
+    cdef:
+        Py_ssize_t n = len(values)
+
+    if loc >= n:
```

Comment:
I suppose you don't even need this `if loc >= n` - the while loop should handle that on its own

Reply:
I think we want the `if loc >= n` so that we only call `np.resize` when necessary.

Reply:
Ah yep you are right - ignore me